File: src/brs/web/api/http/handler/GetAccountsWithName.java
package brs.web.api.http.handler;

import brs.Account;
import brs.BurstException;
import brs.services.AccountService;
import brs.util.Convert;
import brs.web.api.http.ApiServlet;
import brs.web.api.http.common.LegacyDocTag;
import com.google.gson.JsonArray;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;

import javax.servlet.http.HttpServletRequest;
import java.util.Collection;

import static brs.web.api.http.common.Parameters.ACCOUNTS_RESPONSE;
import static brs.web.api.http.common.Parameters.NAME_PARAMETER;

public class GetAccountsWithName extends ApiServlet.JsonRequestHandler {

    private final AccountService accountService;

    public GetAccountsWithName(AccountService accountService) {
        super(new LegacyDocTag[] {LegacyDocTag.ACCOUNTS}, NAME_PARAMETER);
        this.accountService = accountService;
    }

    @Override
    protected
    JsonElement processRequest(HttpServletRequest request) throws BurstException {
        Collection<Account> accounts = accountService.getAccountsWithName(request.getParameter(NAME_PARAMETER));
        JsonArray accountIds = new JsonArray();

        for (Account account : accounts) {
            accountIds.add(Convert.toUnsignedLong(account.id));
        }

        JsonObject response = new JsonObject();
        response.add(ACCOUNTS_RESPONSE, accountIds);
        return response;
    }
}

File: src/main/kotlin/brs/api/http/GetAccountsWithName.kt
package brs.api.http

import brs.api.http.common.JSONResponses.MISSING_NAME
import brs.api.http.common.Parameters.ACCOUNTS_RESPONSE
import brs.api.http.common.Parameters.NAME_PARAMETER
import brs.services.AccountService
import brs.util.convert.toUnsignedString
import brs.util.jetty.get
import com.google.gson.JsonArray
import com.google.gson.JsonElement
import com.google.gson.JsonObject
import javax.servlet.http.HttpServletRequest

/**
 * TODO
 */
internal class GetAccountsWithName internal constructor(private val accountService: AccountService) :
    APIServlet.JsonRequestHandler(arrayOf(APITag.ACCOUNTS), NAME_PARAMETER) {
    override fun processRequest(request: HttpServletRequest): JsonElement {
        val name = request[NAME_PARAMETER] ?: return MISSING_NAME
        val accounts = accountService.getAccountsWithName(name)
        val accountIds = JsonArray()

        for (account in accounts) {
            accountIds.add(account.id.toUnsignedString())
        }

        val response = JsonObject()
        response.add(ACCOUNTS_RESPONSE, accountIds)
        return response
    }
}
